.github/workflows/package-release.yml

```diff
@@ -13,6 +14,12 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
+      - name: Read file contents
```

Comment:
We don't have a release notes file, we use the draft releases page to add the changes and release notes

Reply:
I don't think we can automate that.

Reply:
I don't think we can automate it that way. What this will do is make a release *and* compile the gzp archive every time we push with a tag that matches your specifications.

Reply:
Ok.What format should the RELEASENOTES file take?

Reply:
Markdown

Reply:
And the entire content is copied into the release notes for the tag?

Reply:
That's my understanding, yes

Reply:
Ok. Then I think lets put that file somewhere out of the way, maybe under `.github`. @hairmare, @Robbt what are your thoughts?

Reply:
`action/create-release` does seem to create the actual release which is already done the way I've been releasing in the past. AFAICR there are some existing standards regarding how to format `CHANGELOG.md` that might make sense for us.

I would prefer to keep the actual releasenotes human written though. I've been thinking of moving them to the wiki to make way for autogenerated notes (PR based) in the actual github release. 
A RELEASENOTES.md would work as well but I don't like having to nag people that they should also mention their contribution in the file... maybe we should do consider something this [nag](https://github.com/ponylang/release-notes-reminder-bot-action) action?

We could also use semantic commits and integrate with a semantig release tool that automatically creates tags when we merge to master but IMO this is something to look into after we reach at least beta.

Reply:
I also like the handwritten release notes. Perhaps the nag action is a good idea. Or adding the current milestone to all merged PRs if that can be automated. Then we can do the same as now and go through the merged PRs to create the release notes

Reply:
We can still write our own release notes under the automated method; the Github Action itself doesn't generate the RELEASENOTES file, just pulls from it (like `cat file > newfile` or `myvariable=${cat file}`).

I'm not aware of a plugin that would pull from a Github Wiki but I can keep digging. Keeping basic explainers in static files instead of a CMS or database sounds better to me.

Reply:
I'm happy to merge this with release notes moved to .github/worrkflows. The nag bot is useful, but doesn't need to be part of this PR

Reply:
@paddatrapper Well, this PR also removed the files we need to use Travis CI with. While the tests are running, the Python plugins do have failing tests (which GH Actions still labels as a successful test). Below output is from a test on Ubuntu 18.04:

```
Run sudo bash ./.github/scripts/python-pkg-test.sh
  sudo bash ./.github/scripts/python-pkg-test.sh
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.6.12/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.6.12/x64/lib
..........................................................FFFFFF...
======================================================================
FAIL: tests.replaygain_analyzer_tests.test_mp3_utf8
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 33, in test_mp3_utf8
    check_default_metadata(metadata)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 19, in check_default_metadata
    assert abs(metadata['replay_gain'] - expected_replaygain) < tolerance
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
5.67

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: tests.replaygain_analyzer_tests.test_mp3_dualmono
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 38, in test_mp3_dualmono
    check_default_metadata(metadata)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 19, in check_default_metadata
    assert abs(metadata['replay_gain'] - expected_replaygain) < tolerance
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
5.66

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: tests.replaygain_analyzer_tests.test_mp3_jointstereo
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 43, in test_mp3_jointstereo
    check_default_metadata(metadata)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 19, in check_default_metadata
    assert abs(metadata['replay_gain'] - expected_replaygain) < tolerance
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
5.67

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: tests.replaygain_analyzer_tests.test_mp3_simplestereo
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 48, in test_mp3_simplestereo
    check_default_metadata(metadata)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 19, in check_default_metadata
    assert abs(metadata['replay_gain'] - expected_replaygain) < tolerance
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
5.66

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: tests.replaygain_analyzer_tests.test_mp3_stereo
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 53, in test_mp3_stereo
    check_default_metadata(metadata)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 19, in check_default_metadata
    assert abs(metadata['replay_gain'] - expected_replaygain) < tolerance
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
5.67

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: tests.replaygain_analyzer_tests.test_mp3_mono
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 58, in test_mp3_mono
    check_default_metadata(metadata)
  File "/home/runner/work/libretime/libretime/python_apps/airtime_analyzer/tests/replaygain_analyzer_tests.py", line 19, in check_default_metadata
    assert abs(metadata['replay_gain'] - expected_replaygain) < tolerance
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
5.69

--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 67 tests in 5.804s

FAILED (failures=6)
...........
----------------------------------------------------------------------
Ran 11 tests in 0.157s

OK
```

Reply:
Ah, this looks like Silan issues. Maybe install Silan from the PPA?

Reply:
Also failing tests should equal a failed CI run...

Reply:
The install script should be grabbing Silan from the PPA... I asked it to...

> Also failing tests should equal a failed CI run...

Exactly. Maybe we need to force the test to exit with a different exit code upon an error or failure.

Reply:
btw... I can live with the release parts not working after we merge this, for now our focus should probably be on running tests... I can do the next release manually and/or fix what's not working then if that makes sense.

Reply:
I've made the release creation create a draft release that we can then modify and release ourselves